File: app/src/main/java/com/example/workstation/pdm_se01/WeatherActivity.java
package com.example.workstation.pdm_se01;

import android.content.Intent;
import android.media.Image;
import android.os.Bundle;
import android.support.design.widget.FloatingActionButton;
import android.support.design.widget.Snackbar;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.view.View;
import android.widget.EditText;
import android.widget.ImageView;
import android.widget.TextView;

import com.example.workstation.pdm_se01.DAL.Forecast.Forecast;
import com.squareup.picasso.Picasso;

import org.codehaus.jackson.map.ObjectMapper;
import org.w3c.dom.Text;

import java.io.IOException;
import java.text.SimpleDateFormat;
import java.util.Date;

public class WeatherActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_weather);
        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);
        getSupportActionBar().setDisplayHomeAsUpEnabled(true);

        Intent i = getIntent();
        Date date = (Date) i.getSerializableExtra("Date");

        ObjectMapper mapper = new ObjectMapper();
        Forecast forecast = new Forecast();
        try {
            forecast = mapper.readValue(i.getStringExtra("forecastJSON"), Forecast.class);
        } catch (IOException e) {
            e.printStackTrace();
        }
        fillActivity(forecast,date);
    }
    private void fillActivity(Forecast forecast,Date date) {
        SimpleDateFormat sdf = (SimpleDateFormat) SimpleDateFormat.getDateInstance();
        getSupportActionBar().setTitle(sdf.format(date));

        ImageView myIcon = (ImageView) findViewById(R.id.weatherImage);
        Picasso
                .with(getApplicationContext())
                .load("http://openweathermap.org/img/w/"+forecast.getWeather().get(0).getIcon()+".png")
                .into(myIcon);
        TextView minTemperature = (TextView) findViewById(R.id.minTemperatureWeather);
        minTemperature.setText(Double.toString(forecast.getTemp().getMin()));
        TextView maxTemperature = (TextView) findViewById(R.id.maxTemperatureWeather);
        maxTemperature.setText(Double.toString(forecast.getTemp().getMax()));
        TextView windspeed = (TextView) findViewById(R.id.windSpeedWeather);
        windspeed.setText(Double.toString(forecast.getSpeed()));
        TextView pressure = (TextView) findViewById(R.id.pressure);
        pressure.setText(Double.toString(forecast.getPressure()));
        TextView humidity = (TextView) findViewById(R.id.humidity);
        humidity.setText(Double.toString(forecast.getHumidity()));

    }

}

File: app/src/main/java/com/example/workstation/pdm_se01/activities/WeatherActivity.kt
package com.example.workstation.pdm_se01.activities

import kotlin.jvm.javaClass
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.Toolbar
import android.widget.ImageView
import android.widget.TextView

import com.example.workstation.pdm_se01.model.Forecast.Forecast
import com.example.workstation.pdm_se01.R
import com.squareup.picasso.Picasso

import org.codehaus.jackson.map.ObjectMapper

import java.io.IOException
import java.text.SimpleDateFormat
import java.util.Date

class WeatherActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_weather)
        val toolbar = findViewById(R.id.toolbar) as Toolbar
        setSupportActionBar(toolbar)
        supportActionBar!!.setDisplayHomeAsUpEnabled(true)

        val i = intent
        val date = i.getSerializableExtra("Date") as Date

        val mapper = ObjectMapper()
        var forecast = Forecast()
        try {
            //forecast = mapper.readValue(i.getStringExtra("forecastJSON"), Forecast::class.java )
        } catch (e: IOException) {
            //e.printStackTrace()
        }

        //fillActivity(forecast, date)
    }

    private fun fillActivity(forecast: Forecast, date: Date) {
        val sdf = SimpleDateFormat.getDateInstance() as SimpleDateFormat
        supportActionBar!!.title = sdf.format(date)

        val myIcon = findViewById(R.id.weatherImage) as ImageView
        Picasso.with(applicationContext).load("http://openweathermap.org/img/w/" + forecast.weather[0].icon + ".png").into(myIcon)
        val minTemperature = findViewById(R.id.minTemperatureWeather) as TextView
        minTemperature.text = java.lang.Double.toString(forecast.temp.min)
        val maxTemperature = findViewById(R.id.maxTemperatureWeather) as TextView
        maxTemperature.text = java.lang.Double.toString(forecast.temp.max)
        val wind_speed = findViewById(R.id.windSpeedWeather) as TextView
        wind_speed.text = java.lang.Double.toString(forecast.speed)
        val pressure = findViewById(R.id.pressure) as TextView
        pressure.text = java.lang.Double.toString(forecast.pressure)
        val humidity = findViewById(R.id.humidity) as TextView
        humidity.text = java.lang.Double.toString(forecast.humidity.toDouble())
    }
}
